### ISSUE TITLE
FreeDraw object is automatically reseted 

### ISSUE THREAD

**Issue by fplgusmao** [Wildhoney/Leaflet.FreeDraw#54]
I'm adding a `freeDraw` layer to a `L.layerGroup` on my map. Users could draw their polygons and stuff, then they move to another step of my app. By advancing, the `L.layerGroup`, containing the `freeDraw` layer, is detached from the map, using the `map.removeLayer()` (I keep the reference to the variable, however).

The problem is when users want to get back to the previous step. I add, again, the previous `L.layerGroup` using `someLayerGroup.addTo(someMap)` function, hoping that the `freeDraw` layer keeps the same state as before (maintaining the `polygons` property, for example). However, the whole `freeDraw` object is reseted.

An example:

````javascript
var currentLayerGroup;

userChangedStep = function() {
  savePreviousLayer(currentLayerGroup);
  myMap.removeLayer(currentLayerGroup);
  currentLayerGroup = getStepLayer();

  if(freeDrawStep)
    setupDrawing();

  currentLayerGroup.addTo(myMap);
};

setupDrawing = function() {
  if(!currentLayer.freeDraw) { //if freeDraw is not initialised yet
    currentLayerGroup.freeDraw = new L.FreeDraw(myOptions);
    currentLayerGroup.addLayer(currentLayerGroup.freeDraw);
  }
};
````

Basically, when `userChangedStep()` is called, and the current step is again the "freeDrawStep", I was hoping to reload every polygon that the user previously drew, since the `freeDraw` layer is associated with the current `layerGroup`. However, all the properties on the `freeDraw` object are reseted.

So... Is this a bug or a feature?

**Comment by Wildhoney** [comment]
Unfortunately it's a feature of Leaflet. To solve it, the best way is to keep an array of the polygons by listening to the `Leaflet.FreeDraw` events, and then re-adding those upon adding the layer with the `createPolygon` method.

```javascript
var polygons = [];
freeDraw.on('markers', function getMarkers(eventData) {
    polygons = eventData.latLngs;
});
```

And then later something like the following:

```javascript
polygons.forEach(function(polygon) {
     freeDraw.createPolygon(polygon);
});
```

**Comment by fplgusmao** [comment]
Ok, it seems like a good trick! I'll try and give you the update soon

Thanks a lot

**Comment by fplgusmao** [comment]
Just a question, that `eventData.latLngs` contains every polygon currently on the map? I thought it was only the one that was drawn/deleted/edited

**Comment by Wildhoney** [comment]
Nope, it is every polygon on the map :+1: they're in insertion order though, so the last one is `eventData.latLngs[eventData.latLngs.length - 1]`.

**Comment by fplgusmao** [comment]
I think it's working! Thanks so much, really!

**Comment by Wildhoney** [comment]
No problem! :+1:
